src/package/packageInstall.ts

```diff
@@ -126,19 +135,19 @@ export async function getInstallationStatus(
 export async function waitForPublish(
   connection: Connection,
   subscriberPackageVersionId: string,
-  frequency: number | Duration,
-  timeout: number | Duration,
-  installationKey?: string
+  frequency: number | Duration | undefined,
+  timeout: number | Duration | undefined,
+  installationKey?: string | undefined | Nullable<string>
```

Comment:
frequency, timeout, and installationKey should all be optional for this API.  If frequency or timeout are defined then they should either be a number of Duration.  If installationKey is defined it should be a string.